### PR TITLE
Fix a typing issue in route tree reducers

### PR DESCRIPTION
**Pull request by akalin-keybase**
Also remove unused parameter from configureStore.

### CODE REVIEW

File: shared/reducers/route-tree.js [hunk 1]
```diff
@@ -134,8 +128,7 @@ function routeStateReducer(routeDef, routeState, action) {
 export default function routeTreeReducer(state: Types.State = initialState, action: any): Types.State {
   let {loggedInUserNavigated, routeDef, routeState} = state
   if (action.type === CommonConstants.resetStore) {
-    routeDef = firstRouteDef || initialState
-    // $FlowIssue
+    routeDef = firstRouteDef || initialState.routeDef
```
Comment (akalin-keybase) [inline shared/reducers/route-tree.js]: here was the bug

**Comment by chrisnojima** [comment]
this doens't break the world right? :)

**Comment by akalin-keybase** [comment]
I don't think so, but I'll do some testing before merging!

**Comment by akalin-keybase** [comment]
Did some testing and everything seems to work. Put in breakpoints to make sure that the changed code was exercised.

**Pull request merged**